### PR TITLE
Add rationale and description fields

### LINKED ISSUE #32: Attach rationale / reason to each test
Could do this using test decorators or within the config file

### PR DESCRIPTION
**Pull request by ajvb**
Fixes #32 

r? @g-k

### CODE REVIEW

File: report_json_to_service_json.py [hunk 1]
```diff
@@ -39,6 +39,8 @@
        'reason':  # pytest test outcome reason if any (e.g. resource fetch failed)
        'markers':  # pytest markers on the test e.g. aws service, ruleset
        'metadata':  # additional metadata on the resource being tested
+       'rationale': # (optional) rationale behind the test.
+       'description': # (optional) description of the test
```
Comment (g-k) [inline report_json_to_service_json.py]: Did I forget to add severity here?

File: conftest.py [hunk 2]
```diff
@@ -140,6 +140,10 @@ def get_outcome_and_reason(report, markers, call):
         return report.outcome, None  # passed, failed, skipped
 
 
+def clean_docstring(docstr):
+    return " ".join([word for word in docstr.replace("\n", " ").strip().split(" ") if word != ""])
```
Comment (g-k) [inline conftest.py]: An example of what this is doing might be useful.

File: aws/rds/test_rds_db_security_group_does_not_grant_public_access.py [hunk 3]
```diff
@@ -11,4 +11,8 @@
     rds_db_security_groups(),
     ids=lambda sg: sg['DBSecurityGroupArn'])
 def test_rds_db_security_group_does_not_grant_public_access(rds_db_security_group):
+    """
+    Checks whether any RDS security group allows for inbound
+    access from the public internet
+    """
```
Comment (g-k) [inline aws/rds/test_rds_db_security_group_does_not_grant_public_access.py]: 🎉 for docs!

File: report_json_to_service_json.py (hunk 1 above)
Comment (ajvb) [reply report_json_to_service_json.py]: I believe so

File: conftest.py (hunk 2 above)
Comment (ajvb) [reply conftest.py]: fixed

**Pull request merged**